### PR TITLE
Fixed #29919 -- Fixed RelatedOnlyFieldListFilter crash with reverse relationships.

### PR DESCRIPTION
**Pull request by mgrdcm**
https://code.djangoproject.com/ticket/29919

### CODE REVIEW

**Comment by timgraham** [comment]
Also, I can't reproduce the problem given the information on the ticket. Can you add details there?

**Comment by mgrdcm** [comment]
@timgraham Test case added:  https://github.com/django/django/pull/10606/commits/da4785e82c9586c6f2ab41a0e9e5bc3eeeb8fd1c

**Comment by mgrdcm** [comment]
@auvipy For release notes, will this end up in 2.1.x (4?) bug fix or 2.2?  Apologies that I'm not sure how to determine that.

**Comment by timgraham** [comment]
It doesn't qualify for a backport to 2.1.x per our [supported versions policy](https://docs.djangoproject.com/en/dev/internals/release-process/#supported-versions). Bug fixes that aren't backported don't require release notes.

**Comment by timgraham** [comment]
As in 6d4e5feb79f7eabe8a0c7c4b87f25b1a7f87ca0b, please add tests in `tests/model_fields`. You can rebase and squash commits.

**Comment by mgrdcm** [comment]
Will do!  Might take a day or two.  Appreciate your patience with my first Django bug fix.

**Comment by mgrdcm** [comment]
@timgraham Requests tests have been added, rebasing and squashing too.

File: tests/admin_filters/tests.py [hunk 1]
```diff
@@ -623,6 +627,17 @@ def test_relatedfieldlistfilter_manytomany(self):
         self.assertIs(choice['selected'], True)
         self.assertEqual(choice['query_string'], '?contributors__id__exact=%d' % self.bob.pk)
 
+    def test_relatedfieldlistfilter_reverse_relationships_related_name(self):
+        modeladmin = CustomUserAdminRelatedOnlyRelatedName(User, site)
+
+        request = self.request_factory.get('/')
+        request.user = self.alfred
+
+        try:
```
Comment (timgraham) [inline tests/admin_filters/tests.py]: This test should very the correct results of the Filter rather than just checking that an exception isn't raised.

**Comment by a70babat** [comment]
any update on this? @mgrdcm

**Comment by mgrdcm** [comment]
> any update on this? @mgrdcm

Thank you for the reminder!  I'll work on updating this and adding the requested tests over about the next week.

**Comment by a70babat** [comment]
Thank you so much!

**Comment by nourwolf** [comment]
@mgrdcm a kindly reminder 😊

**Comment by mgrdcm** [comment]
> @mgrdcm a kindly reminder 😊

@nourwolf Thanks for the reminder!  I believe this now has proper tests.

**Comment by felixxm** [comment]
@mgrdcm Thanks for this patch :+1: I moved unrelated tests to a separate commit, rebased from master, simplified tests and added test for `ManyToManyField`.

**Comment by felixxm** [comment]
@mgrdcm Welcome aboard :boat:

**Pull request merged**